templates/kubernetes-cluster.template

```diff
@@ -422,6 +409,59 @@ Resources:
           - LoadBalancerHostname: !GetAtt ApiLoadBalancer.DNSName
             LoadBalancerName: !Ref ApiLoadBalancer
 
+  # IAM role for Lambda function for generating kubeadm token
+  LambdaExecutionRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+        - Effect: "Allow"
+          Principal:
+            Service: ["lambda.amazonaws.com"]
+          Action: "sts:AssumeRole"
+      Path: "/"
+      Policies:
+      - PolicyName: "lambda_policy"
+        PolicyDocument:
+          Version: "2012-10-17"
+          Statement:
+          - Effect: "Allow"
+            Action:
+            - "logs:CreateLogGroup"
+            - "logs:CreateLogStream"
+            - "logs:PutLogEvents"
+            Resource: "arn:aws:logs:*:*:*"
+
+  # Lambda Function for generating the kubeadm token
+  GenKubeadmToken:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Code:
+        ZipFile: |
+          import random
+          import cfnresponse
+          def handler(event, context):
+            if event['RequestType'] == 'Delete':
+              cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
+            if event['RequestType'] == 'Create':
+              token = ("%06x.%016x" % (random.SystemRandom().getrandbits(3*8), random.SystemRandom().getrandbits(8*8)))
```

Comment:
Sure -- give me a chance and I'll try it out.  The previous one was optimized to fit on a command line.